pyiceberg/catalog/rest.py

```diff
@@ -777,9 +778,15 @@ def commit_table(
         identifier = self._identifier_to_tuple_without_catalog(table.identifier)
         table_identifier = TableIdentifier(namespace=identifier[:-1], name=identifier[-1])
         table_request = CommitTableRequest(identifier=table_identifier, requirements=requirements, updates=updates)
+
+        headers = self._session.headers
+        if table_token := table.config.get("token"):
```

Comment:
wdyt about moving this logic to `_response_to_table` and use `self._refresh_token`
https://github.com/apache/iceberg-python/blob/e7de0cceed6e7f4e68c427150b3d3bfd3625536d/pyiceberg/catalog/rest.py#L280

every time we get a `Table` object from the server response, `_refresh_token` will set the session's header
https://github.com/apache/iceberg-python/blob/e7de0cceed6e7f4e68c427150b3d3bfd3625536d/pyiceberg/catalog/rest.py#L549-L551

and subsequent requests should use that automatically

Reply:
I'm not completely following. We want to override the `token` permanently, and only inject it when we commit for a table that has such a token. Otherwise:

```py
t1 = catalog.load_table('t1')
# Now the header is updated with the token for t1
t2 = catalog.load_table('t2')
# Now the header is updated with the token for t2

with t1.transaction():
    # This will commit with the token of t2
```

Did I miss anything?

Reply:
oops yea, I forgot that session is shared!